### PR TITLE
Extend run-tests.ps1 by a warning if submodules are not synced

### PR DESCRIPTION
**Pull request by jongleur1983**
## Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping pull-requests open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules 👮.
- [x] I have provided test coverage for my change (where applicable)

## Description
I'm contributing to ImageSharp since around a week, and I saw 3 people (myself included) stumbling over failing tests due to the submodule for test data being out of sync.

With this commit the run-tests.ps1 test script is extended by the following behaviour:

If something fails during the script run, the submodule status is checked.
A warning is printed if

- any submodule is not initialized or
- any submodule is not synchronized
- git could not be called to check the submodule status.

These warnings contain hints on how to solve that potential issue.

### CODE REVIEW

File: run-tests.ps1 [hunk 1]
```diff
@@ -64,4 +88,22 @@ Invoke-Expression $testRunnerCmd
 
 cd $PSScriptRoot
 
+
+if (0 -ne ([int]$LASTEXITCODE)) {
+  # check submodule status
+  $submoduleStatus = CheckSubmoduleStatus
+  if ([int]$submoduleStatus -eq 1) {
+    # not synced
+	Write-Host -ForegroundColor Yellow "Check if submodules are up to date. You can use 'git submodule update' to fix this";
```
Comment (antonfirsov) [inline run-tests.ps1]: Please use spaces for indentation instead of tabs! (Like the rest of the file.)

Comment (jongleur1983) [reply run-tests.ps1]: resolved, thanks

**Comment by jongleur1983** [comment]
Fixes committed. Sorry for the circumstances - I should have tested it better myself.
Tested is now:
- submodule points to wrong commit (out of sync)
- submodule directory is empty (the test @antonfirsov performed)

and tabs are replaced by spaces of course.

There may still be cases missing, but I'm not sure how to test them.

**Comment by antonfirsov** [comment]
image

Unexpected behavior confirmed.

**Comment by antonfirsov** [comment]
Everything looks good now.

@JimBobSquarePants I hope you don't mind that I'm merging this.

**Comment by JimBobSquarePants** [comment]
@antonfirsov Whenever you think something is good merge away! :smile:

**Pull request merged**